eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/contigalias/ContigAliasService.java

```diff
@@ -0,0 +1,209 @@
+            throw new NoSuchElementException("Not data returned for " + url + " from the contig alias service");
+        }
+        return getTranslatedContig(contigAliasResponse, contigAliasNaming);
+    }
+
+    private AccessionWrapper<ISubmittedVariant, String, Long> createSubmittedVariantAccessionWrapperWithNewContig(
+            AccessionWrapper<ISubmittedVariant, String, Long> accessionWrapper, String newContig) {
+        ISubmittedVariant data = accessionWrapper.getData();
+        ISubmittedVariant dataAfterContigAlias = new SubmittedVariant(data.getReferenceSequenceAccession(),
+                                                                      data.getTaxonomyAccession(),
+                                                                      data.getProjectAccession(),
+                                                                      newContig,
+                                                                      data.getStart(),
+                                                                      data.getReferenceAllele(),
+                                                                      data.getAlternateAllele(),
+                                                                      data.getClusteredVariantAccession(),
+                                                                      data.isSupportedByEvidence(),
+                                                                      data.isAssemblyMatch(),
+                                                                      data.isAllelesMatch(),
+                                                                      data.isValidated(),
+                                                                      data.getCreatedDate());
+        return new AccessionWrapper<>(accessionWrapper.getAccession(), accessionWrapper.getHash(), dataAfterContigAlias);
+}
+
+    private String getTranslatedContig(ContigAliasResponse contigAliasResponse, ContigAliasNaming contigAliasNaming) {
+        ContigAliasChromosome contigAliasChromosome = contigAliasResponse.getEmbedded().getContigAliasChromosomes().get(0);
+        String contig;
+        switch (contigAliasNaming) {
+            case GENBANK_SEQUENCE_NAME:
+                contig = contigAliasChromosome.getName();
+                break;
+            case REFSEQ:
+                contig = contigAliasChromosome.getRefseq();
+                break;
+            case UCSC:
+                contig = contigAliasChromosome.getUcscName();
+                break;
+            case ENA_SEQUENCE_NAME:
+                contig = contigAliasChromosome.getEnaSequenceName();
+                break;
+            case MD5_CHECKSUM:
+                contig = contigAliasChromosome.getMd5checksum();
+                break;
+            case TRUNC512_CHECKSUM:
+                contig = contigAliasChromosome.getTrunc512checksum();
+                break;
+            default:
+                contig = contigAliasChromosome.getGenbank();
```

Comment:
Just to be consistent, I would prefer if we switched all the usage of `Genbank` to `Insdc` and the `Name` to `GenbankSequenceName`

Reply:
This should be consistent across this PR now but not necessarily in contig-alias itself, the latter will probably be addressed in [EVA-2838](https://www.ebi.ac.uk/panda/jira/browse/EVA-2838).